docs/guides/how-to-write-a-gadget.md

```diff
@@ -0,0 +1,141 @@
+# How to write a Gadget?
+
+So you want to write a Gadget for Inspektor Gadget? Great!
+
+A gadget consists of a __Gadget Descriptor__ that provides metadata about your gadget and a __Gadget__ implementation
+(sometimes called tracer in our codebase).
+
+## Gadget Descriptor
+
+The first step to writing a new gadget is providing a _Gadget Descriptor_. Simply create a new struct that implements
+the `gadgets.GadgetDesc` interface, providing all required information, including a name, description, category and so
```

Comment:
Add some links to those interfaces -> https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget@main/pkg/gadgets#GadgetDesc?